test-harness/src/test/groovy/com/netflix/counductor/integration/test/LambdaAndTerminalTaskSpec.groovy

```diff
@@ -0,0 +1,124 @@
+package com.netflix.counductor.integration.test
+
+import com.netflix.conductor.common.metadata.tasks.Task
+import com.netflix.conductor.common.run.Workflow
+import com.netflix.conductor.core.execution.WorkflowExecutor
+import com.netflix.conductor.core.execution.WorkflowSweeper
+import com.netflix.conductor.service.ExecutionService
+import com.netflix.conductor.service.MetadataService
+import com.netflix.conductor.test.util.WorkflowTestUtil
+import com.netflix.conductor.tests.utils.TestModule
+import com.netflix.governator.guice.test.ModulesForTesting
+import spock.lang.Shared
+import spock.lang.Specification
+
+import javax.inject.Inject
+
+@ModulesForTesting([TestModule.class])
+class LambdaAndTerminalTaskSpec extends Specification {
+
+    @Inject
+    ExecutionService workflowExecutionService
+
+    @Inject
+    MetadataService metadataService
+
+    @Inject
+    WorkflowExecutor workflowExecutor
+
+    @Inject
+    WorkflowSweeper workflowSweeper
+
+    @Inject
+    WorkflowTestUtil workflowTestUtil
+
+    @Shared
+    def WORKFLOW_WITH_TERMINATE_TASK = 'test_terminate_task_wf'
+
+    @Shared
+    def WORKFLOW_WITH_TERMINATE_TASK_FAILED = 'test_terminate_task_failed_wf'
+
+    def  WORKFLOW_WITH_LAMBDA_TASK = 'test_lambda_wf'
+
+    def setup() {
+        workflowTestUtil.registerWorkflows(
+                'failure_workflow_for_terminate_task_workflow.json',
+                'terminate_task_completed_workflow_integration_test.json',
+                'terminate_task_failed_workflow_integration.json',
+                'simple_lambda_workflow_integration_test.json'
+        )
+    }
+
+    def cleanup() {
+        workflowTestUtil.clearWorkflows()
+    }
+
+    def "Test workflow with a terminate task when the status is completed"() {
+        given: "workflow input"
+        def workflowInput = new HashMap()
+        workflowInput['a'] = 1
+
+        when: "Start the workflow which has the terminate task"
+        def workflowInstanceId = workflowExecutor.startWorkflow(WORKFLOW_WITH_TERMINATE_TASK, 1,
+                '', workflowInput, null, null, null)
+
+        then: "Ensure that the workflow has started and the first task is in scheduled state"
+        workflowInstanceId
```

Comment:
Is this a null check? If so, we probably need to add it to all the features in all specs.